src/main/java/org/jenkinsci/plugins/pipeline/milestone/MilestoneStep.java

```diff
@@ -37,29 +67,65 @@ public void setLabel(String label) {
         this.label = label;
     }
 
+    @DataBoundSetter
+    public void setOrdinal(Integer ordinal) {
```

Comment:
I would put this in the `@DataBoundConstructor` so you can use

```groovy
milestone 1
```

as a shortcut for

```groovy
milestone ordinal: 1
```